### ISSUE TITLE
[WCAG] Ensure keyboard navigation - LayerList component

### ISSUE THREAD

**Issue by kenchase** [Canadian-Geospatial-Platform/geoview#3212]
### Is there an existing issue for this?

- [x] I have searched the existing issues

### Current Behavior

LayerList items do not behave like buttons when using a keyboard: the spacebar key does not trigger the LayerList items.

Disabled LayerList items receive tab focus

### Expected Behavior

LayerList items should behave like buttons when using a keyboard (triggered by the enter and spacebar keys)

Disabled LayerList items should not receive tab focus
